dvc/fs/data.py

```diff
@@ -19,9 +21,12 @@ class DataFileSystem(FileSystem):
     def _prepare_credentials(self, **config):
         return config
 
-    @wrap_prop(threading.Lock())
-    @cached_property
-    def fs(self):
+    @functools.cached_property
```

Comment:
Ideally, we’d only want to lock per-instance. `functools.cached_property` is equivalent to what we were already doing, it would lock on a class level.

Mypy does not allow wrapping properties, so that’s why I made this change.

Reply:
Also, I think using `wrap_prop` here is wrong, because it locks on every access, whereas we only want to lock once when the method runs.